### PR TITLE
ANA-891: Pricing swap example

### PR DESCRIPTION
**Pull request by jz-finbourne**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs [hunk 1]
```diff
@@ -57,14 +92,14 @@ private void UpsertNamedConventionsToLusid()
         {
             // CREATE the flow conventions and index convention for swap
             string scope = "Conventions";
-            string flowConventionsCode = "GBP-3M";
```
Comment (jz-finbourne) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs:60]: FYI: Changing this to USD-6M for convenience. There is no reason to specifically have GBP-3M. We just want an example to work for some swap.

File: sdk/Lusid.Sdk.Tests/Utilities/DemoInstrumentBase.cs [hunk 2]
```diff
@@ -89,7 +89,7 @@ internal void CallLusidGetValuationEndpoint(LusidInstrument instrument, ModelSel
                 // SimpleStatic pricing is lookup pricing. As such, we upsert a quote.
                 // Note that inside CreatePortfolioAndInstrument, the method TestDataUtilities.BuildInstrumentUpsertRequest books the instrument using "ClientInternal".
                 // Hence upsert a quote using ClientInternal as the instrumentIdType.
-                var quoteRequest = TestDataUtilities.BuildQuoteRequest(scope, instrumentID, QuoteSeriesId.InstrumentIdTypeEnum.ClientInternal, 100m, "USD", TestDataUtilities.EffectiveAt);
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/Utilities/DemoInstrumentBase.cs:92]: Consider wrapping these long lines like this:

var quoteRequest = TestDataUtilities.BuildQuoteRequest(
  scope,
  instrumentID,
  QuoteSeriesId.InstrumentIdTypeEnum.ClientInternal,
  100m,
  "USD",
  TestDataUtilities.EffectiveAt);

Not sure if we have an agreed upon convention, but I typically try to keep lines <120 chars

File: sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs [hunk 3]
```diff
@@ -325,34 +324,35 @@ public static PortfolioHolding BuildCashPortfolioHolding(string currency, string
...
         {
             // different rates as the valuation for some instruments such as fx forwards depend on the difference between the rates curves in two currencies
             var complexMarketData = CreateDiscountCurveHighRates(effectiveAt);
-            var complexMarketDataId = new ComplexMarketDataId(
+            var complexMarketDataId = CreateComplexMarketDataId(effectiveAt, currency);
+            var upsertRequest = new UpsertComplexMarketDataRequest(complexMarketDataId, complexMarketData);
+            return upsertRequest;
+        }
+
+        private static ComplexMarketDataId CreateComplexMarketDataId(DateTimeOffset effectiveAt, string currency)
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs:349]: Consider adding method comments

File: sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs [hunk 4]
```diff
@@ -419,19 +424,22 @@ public static UpsertComplexMarketDataRequest BuildCdsSpreadCurvesRequest(DateTim
...
             );
 
-            var marketData = GetSpreadCurveJsonFromFile("XYZCorp.json");
+            var marketData = GetCdsSpreadCurveJsonFromFile("XYZCorp.json");
             var request = new UpsertComplexMarketDataRequest(marketDataId, marketData);
             return request;
         }
 
-        public static Dictionary<string, UpsertQuoteRequest> BuildQuoteRequest(string scope, string id, QuoteSeriesId.InstrumentIdTypeEnum instrumentIdType, decimal price, string ccy, DateTimeOffset effectiveAt)
+        public static Dictionary<string, UpsertQuoteRequest> BuildQuoteRequest(string id, QuoteSeriesId.InstrumentIdTypeEnum instrumentIdType, decimal price, string ccy, DateTimeOffset effectiveAt)
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs:432]: Consider wrapping this line

File: sdk/Lusid.Sdk.Tests/Utilities/TutorialBase.cs [hunk 5]
```diff
@@ -145,24 +145,24 @@ private void UpsertRecipe(ConfigurationRecipe recipe)
                 .Select(inst => inst.Value.LusidInstrumentId)
                 .ToList();
             
-            // ADD instruments to the portfolio via their LusidInstrumentId
-            var transactionRequest = TestDataUtilities.BuildTransactionRequest(luids, scope, portfolioCode, effectiveFrom);
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/Utilities/TutorialBase.cs:149]: At some point we should consider trying to break this function up. It's getting intense to read

Comment (jz-finbourne) [reply sdk/Lusid.Sdk.Tests/Utilities/TutorialBase.cs:149]: Yeah, this big method would probably get deleted in the future. This whole restructuring/refactoring replicates this method. But we are not quite there yet.

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs [hunk 6]
```diff
@@ -50,15 +85,33 @@ public void InterestRateSwapCreationAndUpsertionExample()
...
+        [TestCase(ModelSelection.ModelEnum.Discounting)]
+        public void InterestRateSwapInlineValuationExample(ModelSelection.ModelEnum model)
+        {
+            var irs = InstrumentExamples.CreateExampleInterestRateSwap();
+            CallLusidGetValuationEndpoint(irs, model);
+        }
+        
+        [TestCase(ModelSelection.ModelEnum.ConstantTimeValueOfMoney)]
+        [TestCase(ModelSelection.ModelEnum.Discounting)]
+        public void EquityOptionPortfolioCashFlowsExample(ModelSelection.ModelEnum model)
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs:111]: Is this in the right file? Why is equity option in here?

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs [hunk 7]
```diff
@@ -94,11 +129,30 @@ private void UpsertNamedConventionsToLusid()
...
         public void InterestRateSwapWithNamedConventionsValuationExample(ModelSelection.ModelEnum model)
         {
             var irs = InstrumentExamples.CreateSwapByNamedConventions();
             UpsertNamedConventionsToLusid();
             CallLusidGetValuationEndpoint(irs, model);
         }
+        
+        [TestCase(ModelSelection.ModelEnum.ConstantTimeValueOfMoney)]
+        [TestCase(ModelSelection.ModelEnum.Discounting)]
+        public void InterestRateSwapWithNamedConventionsInlineValuationExample(ModelSelection.ModelEnum model)
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs:143]: VS is complaining that this method is the exact same as the one above it. Again, how is inline different?

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs [hunk 8]
```diff
@@ -50,15 +85,33 @@ public void InterestRateSwapCreationAndUpsertionExample()
...
+        [TestCase(ModelSelection.ModelEnum.Discounting)]
         public void InterestRateSwapValuationExample(ModelSelection.ModelEnum model)
         {
             var irs = InstrumentExamples.CreateExampleInterestRateSwap();
             CallLusidGetValuationEndpoint(irs, model);
         }
+        
+        [TestCase(ModelSelection.ModelEnum.ConstantTimeValueOfMoney)]
+        [TestCase(ModelSelection.ModelEnum.Discounting)]
+        public void InterestRateSwapInlineValuationExample(ModelSelection.ModelEnum model)
```
[the remark this answers is not in the document]
Comment (jz-finbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs:103]: Rider doesn't tell me this! Thought Rider is meant to be the shinier version! It was just a bad copy and paste, have fixed now.

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs (hunk 6 above)
Comment (jz-finbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs]: Another bad copy and paste ><"

**Review by fightingsleep — APPROVED** [review]

**Pull request merged**